blog/2023-10-29-service-exceptions-and-translations.md

```diff
@@ -0,0 +1,21 @@
+---
+author: Jan Bouwhuis
+authorURL: https://github.com/jbouwh
+authorImageURL: https://avatars.githubusercontent.com/u/7188918?s=96&v=4
+title: Exception handling during and translation support
+---
+
+## Exception handling during service calls
+
+Currently service calls that raise exceptions will log full stack traces. Service calls that fail due to invalid user input don't need a stack trace but would benefit from a helpful error message in the users own language.
+
+To be able to suppress the stack trace in these cases, we introduce `ServiceValidationError` as a new exception type. The `ServiceValidationError` exception can be raised instead of `HomeAssistantError` during the execution of a service call. The error message is to be shown to through the UI, and in the logs. The stack trace is printed at debug level, to support development. For other exceptions that are raised from a service call (including  `HomeAssistantError`) nothing changes and a full stack trace will be printed. [Read more](/docs/core/platform/raising_exceptions).
```

Comment:
```suggestion
To be able to suppress the stack trace in these cases, we introduce `ServiceValidationError` as a new exception type. The `ServiceValidationError` exception can be raised instead of `HomeAssistantError` during the execution of a service call. The error message will show in the UI, and in the logs. The stack trace is printed at debug level, to support development. For other exceptions that are raised from a service call (including  `HomeAssistantError`) nothing changes and a full stack trace is printed. [Read more](/docs/core/platform/raising_exceptions).
```